.github/workflows/generateconfig.yml

```diff
@@ -0,0 +1,133 @@
+name: GenerateConfig
+
+on: workflow_dispatch
+
+jobs:
+  GenerateConfig:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: write
+    env: 
+      IAM_INSTANCE_PROFILE_ARN: ${{secrets.IAM_INSTANCE_PROFILE_ARN}} 
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Install xmllint
+      run: sudo apt-get update && sudo apt-get install libxml2-utils
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with: 
+        role-to-assume: ${{secrets.AMI_GENERATE_CONFIG_ROLE}}
+        aws-region: us-west-2
+    - name: Configure prereqs
+      run: |
+        git config --global user.name "GenerateConfig Action"
+        git config --global user.email "gcaction@github.com"
+        UpdateAl1="false"
+        UpdateAl2="false"
+        UpdateAl2023="false"
+    - name: Check AL1 Updates
+      run: |
+        source_ami_line_nums=(2 3 4 5)
+        source_ami_line_contents=()
+        al1_file_path="./release-al1.auto.pkrvars.hcl"
+        for line_number in "${source_ami_line_nums[@]}"; do
```

Comment:
this seems a bit complicated parsing logic, would it be easier to have our "canonical" variables defined in json and then generate the .hcl file from that?

Reply:
I see. However, since I changed it such that we're looking at the diff except the line with ami_version with Amogh's suggestion, I think it is a bit cleaner now and doesn't include a line by line check.